### DISCUSSION TITLE
Mutations extra attacks damage types should have their corresponding armor piercing capabillities

### DISCUSSION THREAD

**Discussion by leoCottret** [cataclysmbn/Cataclysm-BN#2076] [General]
### Describe the bug

Mutation extra attacks don't have any armor penetration.
Armor will always substract a flat amount from Cut/Pierce/Bash damages.

### Steps To Reproduce

1. Make a character with 90 STR/90 DEX
2. Get the curled horns mutation
3. Spawn a debug monster (no armor)
4. Hit him, curled horns will inflict 50 damages (hard cap for this mutation)
5. Spawn a kevlar hulk (30 bash armor)
6. Hit him, curled horns will inflict 50-30 = 20 damages

### Expected behavior
Damage type should be consistent with weapon damage types, and get the same armor piercing bonus, for several reasons:

1. Consistency. This is clearly hidden knowledge. At some point a player will at least learn that bash and pierce damage have a better armor piercing than cutting, but here for some reasons there are no differences?
2. Balance (with tweaks). One one end, stacking extra mutations attacks will obliterate unarmored enemies even at medium strength level (because they got buffed in BN), but will always be "average at best" against armored ones (you do lose the ability to wear gears).

The bext example would be the 8 tentacles mutations:
- Reminder: this mutation adds 30 encumbrance to hands and remove the ability to wear gloves
- The damage formula is (STR/3+1) * 7 (why not 8?), but the armor will reduce the damage after the "(STR/3+1)" part
- So this will never be triggered against armored opponents -> 
90 STR to start damaging a Kevlar hulk -> ((90/3+1)-30)/7= 7 ("bashing") damage
24 STR to start damaging a Zombie hulk (8 bash armor)

So I propose 2 solutions:
1) Reducing the damage of bash/pierce extra attack mutations and give the armor penetration they should have.
2) Making every extra attack mutations marked as cut damage instead. At least we don't have to balance anything because the balance is not "that" bad, since they got buffed, but this would be more evident that they all have 0 armor penetration. And also add a way for to the average player to know it.

**Comment by RelMayers** [comment]
I'm against proposed solutions.

**Comment by leoCottret** [comment]
A third solution could be simply to add a way for the player to know it.
It's not normal that I had to realise that something wasn't right (tentacle mutation almost never triggered at 14 STR), then do some tests in game to confirm my doubts, then look through the code to learn this counterintuitive mechanic.
This is clearly hidden knowledge.
A tip in the start menu would be a start, and something in "view help" too.
I'm ok to not fix a uncommon behavior because it's already rather balanced and integrated to the game since a long time (I guess), but in that case it should be easily knowable.

**Comment by chaosvolt** [comment]
Solution one seems the most logical, as this actually grants armor penetration. Though I'm not 100% sure about reducing damage for them. As far as effective damage, a bigger issue would be the way the chance of it triggering occurs, in which a basic 8 dexterity massively increases your chance of triggering every single mutation attack. So I'd be in favor of implementing arpen for bash and pierce mutation attacks, but either:
1. Reworking the chance formula with the goal of making bonus attacks closer to the intuitively-expected "1 in X" at 8 dexterity, if that can be done while making greater dexterity values helpful but not immense.
2. Or, rig a way to cap the maximum number of mutation attacks that can proc in one turn. If this is already a thing, it ideally should be lower than the maximum number of bonus attacks you can reasonably rack up (which in my experience tends to be 4).

Regarding idea 1 specifically, two methods come to mind:
1. If the "X in Y" function supports being fed floats and not just integers, it'd be easy to turn the dexterity section of the math into a bit of math that translates a value of 8 into 1, 4 would become 0.5, 12 would become 1.5, etc.
2. Or, knowing that strength-scaling damage for bonus mutation attacks is capped at 15, making dexterity's contribution to chance capped at 15 would allow us to balance with a given range of values instead of having to hope that super-dex'd characters won't be too OP relative to how it triggers for those with average dex.

Adjusting the chance values of mutation attacks to hit desired rarities might still be worth doing either way.
